.github/workflows/reusable_provenance.yaml

```diff
@@ -0,0 +1,171 @@
+          df --human-readable
+
+      - name: Get artifact path
+        id: artifact_path
+        run: |
+          echo "artifact_path=$(tail -1 ${{ inputs.build-config-path }} | grep -oP 'output_path = \K(.*)')" >> $GITHUB_OUTPUT
+
+      - name: Get artifact filename
+        id: filename
+        run: |
+          echo "filename=$(basename ${{ steps.artifact_path.outputs.artifact_path }})" >> $GITHUB_OUTPUT
+
+      - name: Build the artifact
+        run: |
+          ./scripts/build_binary -c ${{ inputs.build-config-path }}
+
+      # Download the Ent CLI and configure a remote with write access using a private API key, if
+      # available (GitHub secrets are not available on forks, see
+      # https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow).
+      # If this step is run from a fork, the API key will be empty, and Ent will work in read-only
+      # mode.
+      # See https://github.com/google/ent
+      - name: Download Ent CLI
+        if: steps.cache-ent.outputs.cache-hit != 'true'
+        env:
+          ENT_URL: https://ent-server-62sa4xcfia-ew.a.run.app
+          ENT_DIGEST: sha256:944a34854a2bf9c5d32f3bffa93885ee1c7ef8ab0f4fcc30898a981050ae4233
+        run: |
+          curl --fail ${ENT_URL}/raw/${ENT_DIGEST} > /usr/local/bin/ent
+          chmod +x /usr/local/bin/ent
+          ent
+          cat <<EOF > ~/.config/ent.toml
+          [[remotes]]
+          name = 'ent-store'
+          url = '${ENT_URL}'
+          write = true
+          api_key = '${{ secrets.ENT_API_KEY }}'
+          EOF
+
+      # Upload artifacts to Ent, where they will be retained and publicly accessible by their hash.
+      # This only applies to `push` events, since it needs access to a valid Ent API key (which is
+      # not available on forks / PRs).
+      - name: Upload to Ent
+        if: github.event_name == 'push'
+        run: |
+          ent put ${{ steps.artifact_path.outputs.artifact_path }}
+
+      # Generate the input to the generate_provenance job.
+      # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+      - name: Generate artifact's base64-encoded hash
+        id: hash
+        run: |
+          # first cd to the directory containing the target binary. This allows us calling sha256
+          # on the target binary instead of having to provide the full path. This is fine for this
+          # experimental version, but has to be cleaned-up once we switch completely to SLSA
+          # provenance generation.
+          cd $(dirname ${{ steps.artifact_path.outputs.artifact_path }})
+          # sha256sum generates sha256 hash for the target binary.
+          # base64 --wrap=0 disables line wrapping, and encodes to base64 and outputs on a single line.
+          echo "hashes=$(sha256sum $(basename ${{ steps.artifact_path.outputs.artifact_path }}) | base64 --wrap=0)" >> $GITHUB_OUTPUT
```

Comment:
How many hashes are there? I thought it would be only one. Also maybe this could be renamed to digest at some point for consistency.